pkg/runner/step.go

```diff
@@ -18,14 +18,49 @@ type step interface {
 	getRunContext() *RunContext
 	getStepModel() *model.Step
 	getEnv() *map[string]string
+	getIfExpression(stage stepStage) string
 }
 
-func runStepExecutor(step step, executor common.Executor) common.Executor {
+type stepStage int
+
+const (
+	stepStagePre stepStage = iota
+	stepStageMain
+	stepStagePost
+)
+
+func (s stepStage) String() string {
+	switch s {
+	case stepStagePre:
+		return "Pre"
+	case stepStageMain:
+		return "Run"
+	case stepStagePost:
+		return "Post"
+	}
+	return "Unknown"
+}
+
+func (s stepStage) getStepName(stepModel *model.Step) string {
+	switch s {
+	case stepStagePre:
+		return fmt.Sprintf("pre-%s", stepModel.ID)
+	case stepStageMain:
+		return stepModel.ID
+	case stepStagePost:
+		return fmt.Sprintf("post-%s", stepModel.ID)
+	}
+	return "unknown"
+}
```

Comment:
Shouldn't this rather go into the step model?

Reply:
This construct creates step names for the step interface defined in this file:

https://github.com/nektos/act/blob/08f22b7c94e547d2f7265fb368e1646e02ddcb1d/pkg/runner/step.go#L13-L22

It has nothing to do with the YAML files which more-or-less defines the model.

Therefore I would favor to keep it here.